client/me/gravatar-updater/index.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import request from 'superagent';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import { getCurrentUser } from 'state/current-user/selectors';
+import Gravatar from 'components/gravatar';
+import { localize } from 'i18n-calypso';
+import * as OAuthToken from 'lib/oauth-token';
+
+export class GravatarUpdater extends Component {
+	constructor() {
+		super();
+		this.handleOnPick = this.handleOnPick.bind( this );
+	}
+
+	static propTypes = {
+		translate: React.PropTypes.func,
+		user: PropTypes.object,
+	};
+
+	handleOnPick( files ) {
```

Comment:
Let's maybe abstract this out as a redux action here. Do you think a good fit could be off of `state/current-user/actions` ?